app/src/main/java/com/example/android/oscarine/signin/SigninViewModel.kt

```diff
@@ -0,0 +1,59 @@
+package com.example.android.oscarine.signin
+
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.example.android.oscarine.network.OscarineApi
+import com.example.android.oscarine.network.models.login_user.LoginUserProperty
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
+
+class SigninViewModel: ViewModel() {
+
+    private val _response = MutableLiveData<String>()
+    val response: LiveData<String>
+        get() = _response
+
+    private val _signinSuccessful = MutableLiveData<Boolean>()
+    val signinSuccessful: LiveData<Boolean>
+        get() = _signinSuccessful
+
+    val username = MutableLiveData<String>()
+
+    val password = MutableLiveData<String>()
+
+    init {
+        _signinSuccessful.value = false
+    }
+
+    private fun postForSigningInUser(userDetails: LoginUserProperty) {
+        OscarineApi.retrofitService.loginUser(userDetails = userDetails).enqueue(object: Callback<Any> {
+            override fun onResponse(call: Call<Any>, response: Response<Any>) {
+                _response.value = response.code().toString()
```

Comment:
Why converting to String? Check for success here and send `"You are now logged in"`

Reply:
@iamareebjamal Should I set `_response.value` to be `"You are now logged in"` upon success?

Reply:
Yes